pkg/dependencies/factory.go

```diff
@@ -290,10 +291,19 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 				return fmt.Errorf("unable to get machine config from file %s: %v", clusterConfigFile, err)
 			}
 
-			tinkerbellIp, err := networkutils.GetLocalIP()
+			logger.V(4).Info("Inferring local Tinkerbell Bootstrap IP from environment")
+			localIp, err := networkutils.GetLocalIP()
```

Comment:
Can you move the infer logic to line 303 so the IP is inferred only if the `tinkerbellBootstrapIp` is empty?

Reply:
That's what I had earlier but I felt Chris asked for the opposite. Lol @chrisdoherty4

Reply:
We need the IP for templating. To avoid lazy state initialization I asked for this to be handled much earlier else we risk having [more] method ordering requirements when using the provider.